cluster-autoscaler/core/static_autoscaler_test.go

```diff
@@ -2266,6 +2266,102 @@ func TestRemoveOldUnregisteredNodesAtomic(t *testing.T) {
 	assert.ElementsMatch(t, wantNames, deletedNames)
 }
 
+func TestZeroOrMaxNodesToDelete(t *testing.T) {
```

Comment:
I'd suggest against adding a test for this internal function. Pls note that the same logic is already covered by one test case in TestStaticAutoscalerInstanceCreationErrors() and TestRemoveOldUnregisteredNodesAtomic().

Your test has an advantage of also testing the non-happy paths, but if you'd like to preserve that, could you add that to one/both of the scenarios above?

Reply:
Done, though I would really like to refactor those tests if I have a second.